plugins/modules/oidc_config.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: oidc_config
+
+author:
+  - Domen Dobnikar (@domen_dobnikar)
+short_description: Handles openID connect configuration.
+description:
+  - Can create or update openID connect configuration.
+version_added: 1.1.0
+extends_documentation_fragment:
+  - scale_computing.hypercore.cluster_instance
+seealso: []
+options:
+  client_id:
+    description:
+      - Provided by authentication server when configuring a new client.
+    type: str
+    required: True
+  shared_secret:
+    description:
+      - Provided by authentication server for client authentication.
+      - Write only.
+    type: str
+  certificate:
+    description:
+      - Plain text of the X.509 PEM-encode certificate.
+    type: str
+  config_url:
+    description:
+      - The openID connect provider configuration information endpoint.
+    type: str
+    required: true
+  scopes:
+    description:
+      - Scopes required to obtain necessary claims.
+    type: str
+    required: true
+notes:
+  - C(check_mode) is not supported.
+"""
+
+EXAMPLES = r"""
+- name: New OIDC config
+  scale_computing.hypercore.oidc_config:
+    client_id: 12345
+    shared_secret: secret_stuff
+    certificate: plain_text_from_x509
+    config_url: https:somwhere.com/this/endpoint
+    scopes: required_scopes
+    state: present
+"""
+
+RETURN = r"""
+record:
+  description:
+    - OIDC config record.
+  returned: success
+  type: dict
+  sample:
+    client_id: 1234
+    certificate: this_certificate
+    config_url: https://somewhere.com/this/endpoint
+    scopes: required_scopes
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors
+from ..module_utils.utils import is_changed
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.oidc import Oidc
+from ..module_utils.typed_classes import TypedOidcToAnsible, TypedDiff
+from ..module_utils.task_tag import TaskTag
+from typing import Union, Tuple
+from time import sleep
+
+
+def ensure_present(
+    module: AnsibleModule,
+    rest_client: RestClient,
+    oidc_obj: Union[Oidc, None],
+    unit: bool,
+) -> Tuple[bool, Union[TypedOidcToAnsible, None], TypedDiff]:
+    before = oidc_obj.to_ansible() if oidc_obj else None
+    oidc_obj_ansible = Oidc.from_ansible(module.params)
+    if oidc_obj is None:
+        task = oidc_obj_ansible.send_create_request(rest_client)
+    else:
+        task = oidc_obj_ansible.send_update_request(rest_client)
+    if not unit:
+        sleep(10)  # Wait for the cluster login (Avoid BAD GATEWAY response)
```

Comment:
No, wait_task requires a response from API and getting a response is the problem here - since we always get `BAD GATEWAY`.
When OIDC config is modified, user gets reconnected to the cluster and causes `BAD GATEWAY` response from the API during the reconnect process.
This is essentially a workaround - but the actual issue should be addressed on the API back-end in my opinion.

Reply:
Then this is best we can do at the moment. We can create waiting-api issue for this.

Reply:
I acknowledge it is reasonable workaround. I accept it.
But magic value of 10 sec - for sure best value depends also on other tasks HyperCore might be executing, and also on how responsive is the OIDC provider.

I played a bit with API module, trying to call the endpoint 2-times in a row. Sometimes I get `502 Bad Gateway` (1st or 2nd call), and sometimes this just works - I get a taskTag back. But more about this later.

Reply:
Not a good solution I agree.
Maybe the sleep value can be based on an already passed parameter timeout?
But even if this value shows to work 100% of the time, we would still get problems running other tasks in parallel to this.